File: rust/src/models/metric_usage.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{billable_metric_id::BillableMetricId, grouped_usage::GroupedUsage};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MetricUsage {
    pub grouped_usage: Vec<GroupedUsage>,

    pub metric_code: String,

    pub metric_id: BillableMetricId,

    pub metric_name: String,

    pub total_value: String,
}

impl MetricUsage {
    pub fn new(
        grouped_usage: Vec<GroupedUsage>,
        metric_code: String,
        metric_id: BillableMetricId,
        metric_name: String,
        total_value: String,
    ) -> Self {
        Self {
            grouped_usage,
            metric_code,
            metric_id,
            metric_name,
            total_value,
        }
    }
}

File: typescript/src/models/metricUsage.ts
// this file is @generated
import { type BillableMetricId, BillableMetricIdSerializer } from "./billableMetricId";
import { type GroupedUsage, GroupedUsageSerializer } from "./groupedUsage";

export interface MetricUsage {
  groupedUsage: GroupedUsage[];

  metricCode: string;

  metricId: BillableMetricId;

  metricName: string;

  totalValue: string;
}

export const MetricUsageSerializer = {
  _fromJsonObject(object: any): MetricUsage {
    return {
      groupedUsage: object["grouped_usage"].map((item: any) =>
        GroupedUsageSerializer._fromJsonObject(item)
      ),
      metricCode: object["metric_code"],
      metricId: BillableMetricIdSerializer._fromJsonObject(object["metric_id"]),
      metricName: object["metric_name"],
      totalValue: object["total_value"],
    };
  },

  _toJsonObject(self: MetricUsage): any {
    return {
      grouped_usage: self.groupedUsage.map((item: any) =>
        GroupedUsageSerializer._toJsonObject(item)
      ),
      metric_code: self.metricCode,
      metric_id: BillableMetricIdSerializer._toJsonObject(self.metricId),
      metric_name: self.metricName,
      total_value: self.totalValue,
    };
  },
};
